internal/controller/object/object.go

```diff
@@ -335,7 +380,10 @@ func getDesired(obj *v1alpha1.Object) (*unstructured.Unstructured, error) {
 	return desired, nil
 }
 
-func getLastApplied(obj *v1alpha1.Object, observed *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+func (e *external) getObservation(obj *v1alpha1.Object, observed *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	if e.applyExtractor != nil {
+		return e.applyExtractor.Extract(observed, "crossplane-provider-kubernetes")
```

Comment:
we should introduce a constact for `"crossplane-provider-kubernetes` that should be used everywhere.